File: ros_ws/src/src/intro_pkg1/src/Subscriber1.cpp
// Header files
#include "ros/ros.h"
#include "intro_pkg1/Equ.h"

// Callback function
void msg_received(const intro_pkg1::Equ::ConstPtr &msg) {
    ROS_INFO("Received, info: %s, value: %f", msg->info.c_str(), msg->value);
}

// Main function
int main(int argc, char **argv) {
    // Initialize node
    ros::init(argc, argv, "MSG_Subscriber");
    // Node Handler
    ros::NodeHandle nodeHandler;
    // Subscriber
    ros::Subscriber subscriberObject = nodeHandler.subscribe<intro_pkg1::Equ>("equation", 2, msg_received);
    // Spinoff
    ros::spin();
    return 0;
}

File: ros_ws/src/src/intro_pkg1/scripts/Subscriber1.py
#!/usr/bin/env python
import rospy
from intro_pkg1.msg import Equ

# Callback function
def MsgReceived(message):
    # Print on screen
    rospy.loginfo("Info: \"{info}\" Value: \"{value}\"".format(info=message.info, value=message.value))

def main():
    # Initialize the node
    rospy.init_node('MSG_Subscriber', anonymous=True)
    # Subscriber
    rospy.Subscriber('equation', Equ, MsgReceived)
    # Spinoff
    rospy.spin()

if __name__ == "__main__":
    try:
        main()
    except rospy.ROSInterruptException:
        rospy.logfatal("Node crashed")
